sources/csv/data.go

```diff
@@ -120,14 +174,25 @@ func SetRowStats(conv *internal.Conv, tables []Table, delimiter rune) {
 				conv.Unexpected(fmt.Sprintf("error processing table %s: file %s is empty.", table.Table_name, filePath))
 				continue
 			}
-			conv.Stats.Rows[table.Table_name] += count - 1
+			conv.Stats.Rows[table.Table_name] += count
 		}
 	}
 }
 
-// getCSVRowCount returns the number of rows in the CSV file.
-func getCSVRowCount(r *csvReader.Reader) (int64, error) {
+// getCSVDataRowCount returns the number of data rows in the CSV file. This excludes the headers if present.
+func getCSVDataRowCount(r *csvReader.Reader, colNames []string) (int64, error) {
 	count := int64(0)
+	srcCols, err := r.Read()
+	if err == io.EOF {
+		return count, nil
+	}
+	if err != nil {
+		return count, fmt.Errorf("can't read csv headers for col names due to: %v", err)
+	}
+	// If the row read was not a header, increase count.
+	if !utils.CheckEqualSets(srcCols, colNames) {
```

Comment:
this is going to be very slow. We should check this only for the first line and proceed without this check for other lines.
Lets check if there's a more efficient way to do this based on number of line breaks or so.

Reply:
Currently, thats what we are doing, only checking for the first line. We read just the first row, check if its a permutation of the table columns. 
If yes, we treat that as a header and move on to the remaining lines where we don't do this matching.
If no, we write off that row because that was a data row.